addons/hr_expense/models/product_product.py

```diff
@@ -24,3 +24,14 @@ def _compute_standard_price_update_warning(self):
                             "There are unposted expenses linked to this category. Updating the category cost will change expense amounts. "
                             "Make sure it is what you want to do."
                         )
+
+    def write(self, vals):
+        result = super().write(vals)
+        if 'standard_price' in vals:
+            expenses_sudo = self.env['hr.expense'].sudo().search([('product_id', 'in', self.ids), ('state', 'in', ['reported', 'draft'])])
```

Comment:
Not sure about the sudo() here :thinking: You wish to disregard multi-company filtering ? Isn't standard_price company-dependent anyway ? Why do we need this, then ?

Reply:
If you don't sudo, you don't modify the expense linked as the search yield nothing and you probably don't have the required rights on expenses. Or am I missing something?

Reply:
Discussed: important in case the user modifying the price of the product does not have the Expense rights. But still we need to change the domain to only search the expenses of the active company (since standard_price is company_dependent)